### PR TITLE
Dev/jdsequeira/us vbs22 20 google flights api

### PR DESCRIPTION
**Pull request by DieS06**

### CODE REVIEW

File: demo/src/main/java/com/project/demo/entity/Airport.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+
+@Table(name = "VO_Airport")
+@Entity
+public class Airport {
+    @Id
+    private Integer airport_id;
+    private String name;
+    private String id;
```
Comment (nefi-cenfo) [inline demo/src/main/java/com/project/demo/entity/Airport.java:13]: What is the propose of this property?

File: demo/src/main/java/com/project/demo/logic/request/GoogleFlightsRequestService.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            throw new Exception("Failed to communicate with the flight API.", e);
+        } catch (Exception e) {
+            throw new Exception("An unexpected error occurred while searching for flights.", e);
+        }
+    }
+
+    private void addQueryParameters(UriComponentsBuilder builder, GoogleFlightsRequestEntity query) throws Exception {
+            if (query.getType() == 1 && query.getReturn_date() != null) {
+                builder.queryParam("return_date", query.getReturn_date());
+            } else if (query.getType() == 2 && query.getReturn_date() == null) {
```
Comment (nefi-cenfo) [inline demo/src/main/java/com/project/demo/logic/request/GoogleFlightsRequestService.java]: Beware of the else if block is doing nothing.

File: demo/src/main/java/com/project/demo/logic/request/OpenAIService.java [hunk 3]
```diff
@@ -21,80 +21,81 @@
...
-    @Value("${openai.api.url}")
-    private String openAIUrl;
-    private final RestTemplate restTemplate;
-    private final ObjectMapper objectMapper;
-    @Autowired
-    public OpenAIService(RestTemplate restTemplate, ObjectMapper objectMapper) {
-        this.restTemplate = restTemplate;
-        this.objectMapper = objectMapper;
-    }
+//    @Value("${openai.api.key}")
```
Comment (nefi-cenfo) [inline demo/src/main/java/com/project/demo/logic/request/OpenAIService.java]: Please uncomment this lines of code before merging.

File: demo/src/main/java/com/project/demo/rest/request/GoogleFlightsRequestController.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Locale;
+
+@RestController
+@RequestMapping("/api/flights")
```
Comment (nefi-cenfo) [inline demo/src/main/java/com/project/demo/rest/request/GoogleFlightsRequestController.java:13]: What is the difference between this and the above one?

**Review by nefi-cenfo — DISMISSED** [review]

File: demo/src/main/java/com/project/demo/entity/Airport.java (hunk 1 above)
Comment (DieS06) [reply demo/src/main/java/com/project/demo/entity/Airport.java:13]: the airport_id is the number identifier and the id is the code or airport that google flights response throws.

**Review by nefi-cenfo — DISMISSED** [review]

**Review by kendallduranv — APPROVED** [review]

**Review by mur-tz — APPROVED** [review]
For the next iteration, we may merge the  two controllers on this PR, since they are similar and I think you only use one controller for the request one

**Pull request merged**